blkin-lib/zipkin_c.c

```diff
@@ -92,6 +92,20 @@ int blkin_init_new_trace(struct blkin_trace *new_trace, const char *service,
 	return res;
 }
 
+int blkin_init_root_info(struct blkin_trace_info *trace_info)
+{
+	int res;
+	if (!trace_info) {
+		res = -EINVAL;
+		goto OUT;
+	}
+	trace_info->span_id = trace_info->trace_id = random_big();
+	trace_info->parent_span_id = 0;
+
+OUT:
+	return res;
+}
+
```

Comment:
let's call this `blkin_init_trace_info()` and have it return void (don't bother checking for `!trace_info`)

we don't want span_id and trace_id to be the same, assign them separate `random_big()`s

it should do the same thing that `blkin_init_new_trace()` is doing to its `new_trace->info`, so we can modify `blkin_init_new_trace()` to call `blkin_init_trace_info(&new_trace->info)` instead